Feat/vpc module

munaai opened the pull request:
This PR introduces the following changes to the VPC module:
	•	Adds a fully functional AWS VPC Terraform module with documented inputs and outputs.
	•	Supports configurable public and optional private subnets, an Internet Gateway, and an optional NAT Gateway.
	•	Adds optional VPC Flow Logs with CloudWatch Logs integration.
	•	Introduces a working VPC example used for validation and testing.
	•	Adds Terratest coverage to validate VPC provisioning and teardown.
	•	Aligns the module and examples structure with existing AWS modules.
Related Issue
Closes #3 – Create VPC module skeleton

Testing
	•	Successful Terratest execution validating terraform init, apply, and destroy
	•	Terraform plan verified for the VPC example
	•	Module outputs validated during Terratest execution

Screenshot
<img width="750" height="86" alt="Screenshot 2025-12-20 at 19 04 28" src="https://github.com/user-attachments/assets/12c27608-cb4c-4802-9334-456a39377ee2" />

munaai committed:
feat: structure of the VPC module skeleton

munaai committed:
completed VPC module with example

munaai committed:
completed VPC module with example

munaai committed:
module updated with vpc flow logs

munaai committed:
formatting added and README updated

munaai committed:
README updated

munaai committed:
README updated

munaai committed:
README updated

munaai committed:
Enable private subnet option added

munaai committed:
README updated

munaai committed:
README updated

munaai committed:
Add VPC example outputs and Terratest

munaai committed:
Folder restructure and previous versions added

munaai committed:
versions updated

munaai committed:
file path updated

JunedConnect commented:
Looks good.

Some points:
- For the readme, we could mention that by default public subnets are created explictely? This is personal preference so if you dont think it's needed then that's fine too

- For the tags, rather than tagging each resource itself nad using local tags, can we implement the same that we did for the EC2 and ECS module i.e.:
provider "aws" {
  default_tags {
    tags = {
      ManagedBy = "Terraform"
      Module    = "CoderCo community module - vpc"
    }
  }
}
This will essentially tag every aws resource automatically, rather than having to do it manually.


- For the provider block, can we remove the hardcoded region, we don't need this since the module will auto use whatever the user has set, overwise the user will wonder why their resources are being created in a different AZ since not everyone may use eu-west-2. I have left a comment on the specific code (with the comment "this")  so it helps you pinpoint what I'm referring to.

JunedConnect commented on examples/aws/vpc/main.tf:
```diff
@@ -0,0 +1,32 @@
+    }
+  }
+}
+provider "aws" {
```
this

munaai committed:
tagging sorted out and harcoded region removed

munaai commented:
•	Updated the README  to clarify default public subnet creation
        •      Removed the hardcoded AWS region
	•	Removed local tag merging and now apply only Name tags per resource where appropriate.

munaai committed:
.gitignore updated

munaai committed:
chore: remove terraform lock files and ignore them

munaai commented:
Merged to Dev
